File: src/modules/common/components/decorators/maxLength.ts
import decoratorHelper from "../../helpers/decoratorHelper"

export function maxLength(messageKey: string, max: number) {
    return function (target: any, propertyKey: string) {
        return decoratorHelper.define(target, propertyKey, messageKey, isValid);
    }

    function isValid(value: any): boolean {
        return !!value && value.length <= max;
    }
}

File: Rest/ExamERP/ExamERP.Common/Attributes/MaxLength.cs
namespace ExamERP.Common.Attributes
{
    public class MaxLength : BaseAttribute
    {
        public int Max { get; set; }
        public MaxLength(string messageKey, int maxLength) : base(messageKey)
        {
            this.Max = maxLength;
        }
        public override bool IsValid(object value)
        {
            return value.ToString().Length < this.Max;
        }
    }
}
